hws/src/dr_action.py

```diff
@@ -3,170 +3,64 @@
 
     return (2, [action_pretiffy(action)])
 
 def dr_action_remove_by_size_parser(action_arr, index):
     action_dw_0 = action_arr[index]
     action_dw_1 = action_arr[index + 1]
-    action = {"type" : "Remove by size"}
-    start_anchor = int(action_dw_0[10 : 16], 2)
-    field = modify_pattern_anchor_dic.get(start_anchor)
-    action["start_anchor"] = field if field != None else start_anchor
-    action["outer_l4_removed"] = int(action_dw_0[16 : 17], 2)
-    action["start_offset"] = int(action_dw_0[18 : 25], 2)
-    action["size"] = int(action_dw_0[26 : 32], 2)
+    action = dr_parse_remove_by_size_action(action_dw_0, action_dw_1)
 
     return (1, [action_pretiffy(action)])
 
 def dr_action_remove_header2header_parser(action_arr, index):
     action_dw_0 = action_arr[index]
     action_dw_1 = action_arr[index + 1]
-    action = {"type" : "Remove header2header"}
-    start_anchor = int(action_dw_0[10 : 16], 2)
-    end_anchor = int(action_dw_0[18 : 24], 2)
-    field = modify_pattern_anchor_dic.get(start_anchor)
-    action["start_anchor"] = field if field != None else start_anchor
-    field = modify_pattern_anchor_dic.get(end_anchor)
-    action["end_anchor"] = field if field != None else end_anchor
-    action["decap"] = int(action_dw_0[28 : 29], 2)
-    action["vni_to_cqe"] = int(action_dw_0[29 : 30], 2)
-    action["qos_profile "] = int(action_dw_0[30 : 32], 2)
+    action = dr_parse_remove_header2header_action(action_dw_0, action_dw_1)
 
     return (1, [action_pretiffy(action)])
 
 def dr_action_insert_inline_parser(action_arr, index):
     action_dw_0 = action_arr[index]
     action_dw_1 = action_arr[index + 1]
-    action = {"type" : "Insert inline"}
-    start_anchor = int(action_dw_0[10 : 16], 2)
-    field = modify_pattern_anchor_dic.get(start_anchor)
-    action["start_anchor"] = field if field != None else start_anchor
-    end_anchor = int(action_dw_0[18 : 24], 2)
-    field = modify_pattern_anchor_dic.get(end_anchor)
-    action["end_anchor"] = field if field != None else end_anchor
-    action["insert_data_inline"] = int(action_dw_1[0 : 32], 2)
+    action = dr_parse_insert_inline_action(action_dw_0, action_dw_1)
 
     return (2, [action_pretiffy(action)])
 
 def dr_action_insert_pointer_parser(action_arr, index):
     action_dw_0 = action_arr[index]
     action_dw_1 = action_arr[index + 1]
-    action = {"type" : "Insert pointer"}
-    start_anchor = int(action_dw_0[10 : 16], 2)
-    field = modify_pattern_anchor_dic.get(start_anchor)
-    action["start_anchor"] = field if field != None else start_anchor
-    end_anchor = int(action_dw_0[18 : 24], 2)
-    field = modify_pattern_anchor_dic.get(end_anchor)
-    action["end_anchor"] = field if field != None else end_anchor
-    action["size"] = int(action_dw_0[24 : 29], 2)
-    action["attributes"] = int(action_dw_0[29 : 32], 2)
-    action["pointer"] = int(action_dw_1[0 : 32], 2)
+    action = dr_parse_insert_by_pointer_action(action_dw_0, action_dw_1)
```

Comment:
All this refactoring is nice. It has nothing to do though with pattern/args parsing.
You missed adding all those functions - please add them and do all this in a separate patch.

Reply:
i have changed the parsing function to be used in the regular copy/set/add/... actions when set in the STE directly, and to also be used when copy/set/add/... are part of pattern and argument.
i have but the functions in hws/src/dr_common_functions.py